unified_planning/io/anml_writer.py

```diff
@@ -0,0 +1,329 @@
+                raise NotImplementedError
+
+        for t in self.problem.user_types: # Define objects
+            objects: List['unified_planning.model.Object'] = list(self.problem.objects(t))
+            obj_names = [o.name for o in self.problem.objects(t)]
+            if len(objects) > 0:
+                out.write(f'instance {_get_mangled_type_name(t, types_mapping)} {", ".join(obj_names)};\n')
+
+        for fe, v in self.problem.initial_values.items():
+            out.write(f'[ start ] {converter.convert(fe)} := {converter.convert(v)};\n')
+
+        for ti, el in self.problem.timed_effects.items():
+            for e in el:
+                out.write(_convert_effect(e, converter, ti))
+
+        for g in self.problem.goals:
+            out.write(f'[ end ] {converter.convert(g)};\n')
+
+        for i, gl in self.problem.timed_goals.items():
+            for g in gl:
+                out.write(f'{_convert_anml_interval(i)} {converter.convert(g)};\n')
+
+    def print_problem(self):
+        '''Prints to std output the ANML problem.'''
+        self._write_problem(sys.stdout)
+
+    def get_problem(self) -> str:
+        '''Returns the ANML problem.'''
+        out = StringIO()
+        self._write_problem(out)
+        return out.getvalue()
+
+    def write_problem(self, filename: str):
+        '''Dumps to file the ANML problem.'''
+        with open(filename, 'w') as f:
+            self._write_problem(f)
+
+def _convert_effect(effect: 'up.model.Effect', converter: ConverterToANMLString, timing: 'up.model.Timing' = None) -> str:
+    results: List[str] = []
+    if effect.is_conditional():
+        results.append(f'when [ all ] {converter.convert(effect.condition)}\n{{')
+    if timing is not None:
+        results.append(f'[ {_convert_anml_timing(timing)} ] ')
+    else:
+        results.append('[ start ] ')
+    results.append(converter.convert(effect.fluent))
+    if effect.is_assignment():
+        results.append(' := ')
+    elif effect.is_increase():
+        results.append(' :+= ')
+    elif effect.is_decrease():
+        results.append(' :-= ')
+    else:
+        raise NotImplementedError
+    results.append(f'{converter.convert(effect.value)};\n')
+    if effect.is_conditional():
+        results.append('}\n')
+    return ''.join(results)
+
+def _convert_anml_timing(timing: 'up.model.Timing') -> str:
+    time = 'start' if timing.is_from_start() else 'end'
+    if timing.delay > 0:
+        return f'{time} + {str(timing.delay)}'
+    elif timing.delay == 0:
+        return time
+    else: #timing.delay < 0
+        return f'{time} - {str(timing.delay * (-1))}'
+
+def _convert_anml_interval(interval: 'up.model.TimeInterval') -> str:
+    left_bracket = '(' if interval.is_left_open() else '['
+    right_bracket = ')' if interval.is_left_open() else ']'
+    return f'{left_bracket} {_convert_anml_timing(interval.lower)}, {_convert_anml_timing(interval.upper)} {right_bracket}'
+
+def _is_valid_anml_name(name: str) -> bool:
```

Comment:
ANML has some keywords that are not valid names (e.g. "action") those should be handled. Moreover, I do not see why standard types such as "boolean" should enter here: I was expecting only user-defined types, and names of entities such as fluents should enter here and `_get_anml_valid_name`